Dockerfile

```diff
@@ -7,11 +7,17 @@ RUN apk update && apk add py3-psycopg2 musl-dev \
     nginx supervisor git \
     openssl ca-certificates \
     gcc postgresql-dev \
- && pip install --upgrade pip \
- && pip install virtualenv
+ && pip install --upgrade pip
 
 RUN         pip install -r /app/requirements.txt
-ADD         . /app
+ADD         dataservice /app
```

Comment:
Wondering if you really meant to use ADD vs COPY? See https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#add-or-copy

You can have just one ADD/COPY statement to copy all of the files/dirs you need. This will avoid creating a bunch of unnecessary image layers. (Maybe also consider refactoring the rest of the Dockerfile to use minimal RUN steps for the same reason)

Reply:
I did mean to do that, I think it should be right now.